pkg/registry/core/service/ipallocator/controller/repair.go

```diff
@@ -105,7 +105,7 @@ func (c *Repair) runOnce() error {
 	// the service collection. The caching layer keeps per-collection RVs,
 	// and this is proper, since in theory the collections could be hosted
 	// in separate etcd (or even non-etcd) instances.
-	list, err := c.registry.ListServices(ctx, nil)
```

Comment:
We just should avoid passing nil - it seems there are just few such places in the code and we should forbid it.